store/ZkLobbyServer.js

```diff
@@ -225,6 +225,13 @@ module.exports = function(){ return Reflux.createStore({
 			} else {
 				this.login();
 			}
+			var zklsVersion = msg.Version;
+			if (msg.Engine){
+				var engine = msg.Engine;
+				engine.match(/^[0-9.]+-[0-9]+-g[a-f0-9]+$/) && (engine += ' develop');
```

Comment:
brutal: now this logic is repeated four times, and one of them is a short-circuit